src/SipiImage.cpp

```diff
@@ -760,6 +763,109 @@ namespace Sipi {
 
 #undef POSITION
 
+    bool SipiImage::scaleFast(size_t nnx, size_t nny) {
+        auto xlut = shttps::make_unique<size_t[]>(nnx);
+        auto ylut = shttps::make_unique<size_t[]>(nny);
+
+        for (size_t i = 0; i < nnx; i++) {
+            xlut[i] = (size_t) (i * (nx - 1) / (nnx - 1) + 0.5);
+        }
+        for (size_t i = 0; i < nny; i++) {
+            ylut[i] = (size_t) (i * (ny - 1) / (nny - 1 ) + 0.5);
+        }
+
+        if (bps == 8) {
+            byte *inbuf = pixels;
+            byte *outbuf = new byte[nnx * nny * nc];
+            for (size_t y = 0; y < nny; y++) {
+                for (size_t x = 0; x < nnx; x++) {
+                    for (size_t k = 0; k < nc; k++) {
+                        outbuf[nc * (y * nnx + x) + k] = inbuf[nc * (ylut[y] * nx + xlut[x]) + k];
+                    }
+                }
+            }
+            pixels = outbuf;
+            delete[] inbuf;
+        } else if (bps == 16) {
+            word *inbuf = (word *) pixels;
+            word *outbuf = new word[nnx * nny * nc];
+            for (size_t y = 0; y < nny; y++) {
+                for (size_t x = 0; x < nnx; x++) {
+                    for (size_t k = 0; k < nc; k++) {
+                        outbuf[nc * (y * nnx + x) + k] = inbuf[nc * (ylut[y] * nx + xlut[x]) + k];
+                    }
+                }
+            }
+            pixels = (byte *) outbuf;
+            delete[] inbuf;
+
+        } else {
+            return false;
+
+        }
+
+        nx = nnx;
+        ny = nny;
+        return true;
+    }
+    /*==========================================================================*/
+
+
+    bool SipiImage::scaleMedium(size_t nnx, size_t nny) {
+        auto xlut = shttps::make_unique<float[]>(nnx);
+        auto ylut = shttps::make_unique<float[]>(nny);
+
+        for (size_t i = 0; i < nnx; i++) {
+            xlut[i] = (float) (i * (nx - 1)) / (float) (nnx - 1);
+        }
+        for (size_t j = 0; j < nny; j++) {
+            ylut[j] = (float) (j * (ny - 1)) / (float) (nny - 1);
+        }
+
+        if (bps == 8) {
+            byte *inbuf = pixels;
+            byte *outbuf = new byte[nnx * nny * nc];
```

Comment:
Could we make those byte some kind of smart pointer? to avoid the `delete[] pixels;` line in the distructor